cmd/process-agent/api/server.go

```diff
@@ -0,0 +1,55 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+
+	ddconfig "github.com/DataDog/datadog-agent/pkg/config"
+	settingshttp "github.com/DataDog/datadog-agent/pkg/config/settings/http"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/gorilla/mux"
+)
+
+func setupHandlers(r *mux.Router) {
+	r.HandleFunc("/config", settingshttp.Server.GetValue)
+	r.HandleFunc("/config", settingshttp.Server.GetFull("")).Methods("GET")
+	r.HandleFunc("/config/list-runtime", settingshttp.Server.ListConfigurable).Methods("GET")
+	r.HandleFunc("/config/{setting}", settingshttp.Server.GetValue).Methods("GET")
+	r.HandleFunc("/config/{setting}", settingshttp.Server.SetValue).Methods("POST")
+}
+
+// StartServer starts the config server
+func StartServer() error {
+	// Set up routes
+	r := mux.NewRouter()
+	setupHandlers(r)
+
+	addr, err := getIPCAddressPort()
+	if err != nil {
+		return err
+	}
+	log.Infof("API server listening on %s", addr)
+
+	srv := &http.Server{
+		Handler: r,
+		Addr:    addr,
+	}
+
+	go func() {
+		err := srv.ListenAndServe()
+		if err != nil {
+			_ = log.Error(err)
+		}
+	}()
+	return nil
+}
+
+// getIPCAddressPort returns a listening connection
+func getIPCAddressPort() (string, error) {
+	address, err := ddconfig.GetIPCAddress()
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("%v:%d", address, ddconfig.Datadog.GetInt("process_config.cmd_port")), nil
```

Comment:
Not consider using https://golang.org/pkg/net/#JoinHostPort

Reply:
Oh! I didn't know this was a thing! Thanks for pointing that out to me :)